File: example-cashroulette-java/src/main/java/com/avatye/example/crj/MainActivity.java
package com.avatye.example.crj;

import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.avatye.cashroulette.CashRouletteSDK;
import com.avatye.cashroulette.INotificationStatus;
import com.avatye.example.crj.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {


    private ActivityMainBinding binding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        /** view binding */
        binding = ActivityMainBinding.inflate(LayoutInflater.from(this));
        setContentView(binding.getRoot());

        /** view profile */
        viewProfile();

        /** set profile */
        binding.buttonAuth.setOnClickListener(v -> {
            if (binding.userId.getText() == null || TextUtils.isEmpty(binding.userId.getText())) {
                Toast.makeText(this, "아이디를 입력하세요", Toast.LENGTH_SHORT).show();
            } else {
                setProfile();
            }
        });

        /** start cash roulette */
        binding.buttonEnter.setOnClickListener(v -> {
            CashRouletteSDK.start(this);
        });

        /** refresh ticket balance */
        binding.ticketBalance.setOnClickListener(v -> {
            ticketCondition();
        });

        /** refresh ticket condition */
        binding.ticketCondition.setOnClickListener(v -> {
            ticketCondition();
        });

        CashRouletteSDK.setNotificationStatus(this, true, new INotificationStatus() {
            @Override
            public void onStatus(boolean isActive, int code, @NonNull String reason) {
                Log.d("Notification", "isActive:" + isActive + ", code:" + code + ", reason:" + reason);
            }
        });
    }


    void viewProfile() {
        String appUserID = CashRouletteSDK.getAppUserID();
        if (!TextUtils.isEmpty(appUserID)) {
            binding.wrapAuth.setVisibility(View.GONE);
            binding.wrapProfile.setVisibility(View.VISIBLE);
            binding.profileId.setText("ID : " + appUserID);
            ticketCondition();
        } else {
            binding.wrapAuth.setVisibility(View.VISIBLE);
            binding.wrapProfile.setVisibility(View.GONE);
        }
    }


    void ticketCondition() {
        CashRouletteSDK.getTicketCondition((balance, condition) -> {
            binding.ticketBalance.setText("총 티켓 : " + balance);
            binding.ticketCondition.setText("받을 수 있는 티켓 : " + condition);
        });
    }


    void setProfile() {
        String appUserID = binding.userId.getText().toString();
        CashRouletteSDK.setAppUserID(appUserID);
        viewProfile();
    }
}

File: example-cashroulette-kotlin/src/main/java/com/avatye/example/crk/MainActivity.kt
package com.avatye.example.crk

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.avatye.cashroulette.CashRouletteSDK
import com.avatye.cashroulette.ITicketCount
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewProfile()

        /** set profile */
        button_auth.setOnClickListener {
            if (user_id.text.isNullOrEmpty()) {
                Toast.makeText(this, "아이디를 입력하세요", Toast.LENGTH_SHORT).show()
            } else {
                setProfile()
            }
        }

        /** start cash roulette */
        button_enter.setOnClickListener {
            CashRouletteSDK.start(this)
        }

        /** refresh ticket balance */
        ticket_balance.setOnClickListener {
            ticketCondition()
        }

        /** refresh ticket condition */
        ticket_condition.setOnClickListener {
            ticketCondition()
        }
    }


    private fun viewProfile() {
        val appUserID = CashRouletteSDK.getAppUserID()
        if (appUserID.isNotEmpty()) {
            wrap_auth.visibility = View.GONE
            wrap_profile.visibility = View.VISIBLE
            profile_id.text = "ID : ${appUserID}"
            ticketCondition()
        } else {
            wrap_auth.visibility = View.VISIBLE
            wrap_profile.visibility = View.GONE
        }
    }


    private fun ticketCondition() {
        CashRouletteSDK.getTicketCondition(listener = object : ITicketCount {
            override fun callback(balance: Int, condition: Int) {
                ticket_balance.text = "총 티켓 : $balance"
                ticket_condition.text = "받을 수 있는 티켓 : $condition"
            }
        })
    }


    private fun setProfile() {
        val appUserID = user_id.text.toString()
        CashRouletteSDK.setAppUserID(appUserID)
        viewProfile()
    }
}